app/src/main/java/com/geeksville/mesh/ui/NodeInfo.kt

```diff
@@ -87,7 +87,12 @@ fun NodeInfo(
     val (textColor, nodeColor) = thatNodeInfo.colors
 
     val position = thatNodeInfo.position
-    val hwInfoString = thatNodeInfo.user?.hwModelString
+    val hwInfoString = thatNodeInfo.user?.hwModelString ?: MeshProtos.HardwareModel.UNSET.name
+    val roleName = thatNodeInfo.user?.role?.let { role ->
+        DeviceConfig.Role.forNumber(role)?.name
```

Comment:
we won't be able to set `Role.UNRECOGNIZED` as default for unknown users in [MeshService](https://github.com/meshtastic/Meshtastic-Android/blob/24886994a6afc1a443b92e2630903612403158f8/app/src/main/java/com/geeksville/mesh/service/MeshService.kt#L373-L379) so let's use an if/else block to check `isUnknownUser` and assign `Role.UNRECOGNIZED.name` when its true.